src/gas/script-api.js

```diff
@@ -29,110 +34,144 @@ class ScriptApi {
+        location.reload();
+      })
+      .catch((err) => {
+        showLog(err.message, LEVEL_ERROR);
+      });
   }
 
   getGasCode() {
     return this.updateToken()
-    .then(() => {
-      return new Promise((resolve, reject) => {
-        $.getJSON(
-          `${API_URL}/${context.id}/content`,
-          { access_token: context.gapiToken }
-        )
-        .then(resolve)
-        .fail(reject)
+      .then(() => {
+        return new Promise((resolve, reject) => {
+          $.getJSON(
+              `${API_URL}/${getId()}/content`, {
+                access_token: context.gapiToken
+              }
+            )
+            .then(resolve)
+            .fail(reject)
+        })
       })
-    })
-    .then(content => {
-      const code = content.files.reduce((hash, elem) => {
+      .then(content => {
+        const code = content.files.reduce((hash, elem) => {
           if (elem) {
             let type;
-            switch(elem.type) {
-              case "HTML" :
+            switch (elem.type) {
+              case "HTML":
                 type = ".html"
                 break;
-              case "JSON" :
+              case "JSON":
                 type = ".json"
                 break;
-              case "SERVER_JS" :
-                type = context.config.filetype
+              case "SERVER_JS":
+                type = getConfig().filetype
                 break;
             }
             hash[elem.name + type] = elem.source;
           }
           return hash;
         }, {});
-      return code;
-    });
+        return code;
+      });
   }
 
   updateGas() {
     return new Promise((resolve, reject) => {
-      $.ajax({
-        url: `${API_URL}/${context.id}/content?access_token=${context.gapiToken}`,
-        method: 'PUT',
-        contentType: 'application/json',
-        data: JSON.stringify({
-          files: files
-        })
+        $.ajax({
+            url: `${API_URL}/${getId()}/content?access_token=${context.gapiToken}`,
+            method: 'PUT',
+            contentType: 'application/json',
+            data: JSON.stringify({
+              files: files
+            })
+          })
+          .then(resolve)
+          .fail(reject)
       })
-      .then(resolve)
-      .fail(reject)
-    })
-    .then(() => {
-      showAlert('Successfully pulled from scm');
-      location.reload();
-    })
-    .catch((err) => {
-      showAlert(err.message, LEVEL_ERROR);
-    });
+      .then(() => {
+        showLog('Successfully pulled from scm');
+        location.reload();
+      })
+      .catch((err) => {
+        showLog(err.message, LEVEL_ERROR);
+      });
   }
 
-
   updateToken() {
-    return new Promise((resolve, reject) => {
-      chrome.runtime.sendMessage({ cmd: 'login', interactive: false }, token => {
-        context.gapiToken = token;
-        resolve();
+    if (context.gapiRefreshToken) {
+      return new Promise((resolve, reject) => {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/leonhartX/gas-github/blob/083dcd77ddb6869d2b7dc2cdf2f7452c410d4901/src/gas/script-api.js#L141"></span>‘reject’ is defined but never used.</td>
    </tr>
  </tbody>
</table>